Modules/Devices/src/Devices.Infrastructure/InfrastructureErrors.cs

```diff
@@ -10,4 +10,10 @@ public static InfrastructureError InvalidPushNotificationConfiguration(List<stri
         return new InfrastructureError("error.platform.unsupportedAppId",
             $"The given app id is not supported. Supported app ids are: {supportedAppIdsString}.");
     }
+
+    public static InfrastructureError ConcurrentDeviceRegistrationCreationViolatesPrimaryKeyConstraint(string deviceId)
```

Comment:
I would rather go with `UniqueKeyViolation` to make it reusable. And adapt the error message accordingly.